src/rqt_graph/dotcode.py

```diff
@@ -498,41 +654,98 @@ def generate_dotgraph(self,
                                              rankdir=orientation)
 
         ACTION_TOPICS_SUFFIX = '/action_topics'
-        namespace_clusters = {}
-        for n in (nt_nodes or []) + [action_prefix + ACTION_TOPICS_SUFFIX for (action_prefix, _) in action_nodes.items()]:
+        IMAGE_TOPICS_SUFFIX = '/image_topics'
+
+        namespace_clusters = self._populate_node_graph(cluster_namespaces_level, (nt_nodes or [])
+                                    + [action_prefix + ACTION_TOPICS_SUFFIX for (action_prefix, _) in action_nodes.items()]
+                                    + [image_prefix + IMAGE_TOPICS_SUFFIX for (image_prefix, _) in image_nodes.items()]
+                                    + nn_nodes if nn_nodes is not None else [],
+                                    dotcode_factory, dotgraph, rank, orientation, simplify)
+
+        for n in nt_nodes or []:
+             # cluster topics with same namespace
+             if (cluster_namespaces_level > 0 and
+                n.strip().count('/') > 1 and
+                len(n.strip().split('/')[1]) > 0):
+                if (n.count('/') <= cluster_namespaces_level):
+                    namespace = str('/'.join(n.strip().split('/')[:-1]))
+                else:
+                    namespace = '/'.join(n.strip().split('/')[:cluster_namespaces_level+1])
+                if namespace not in namespace_clusters:
+                    print("Namespace '"+namespace+"' not found.")
```

Comment:
When does this happen? Is that an error?

Same below.

Reply:
There were cases where `namespace` was not in `namespace_clusters` during testing; this line will be removed.